provider/baked/src/export.rs

```diff
@@ -254,11 +254,21 @@ impl BakedExporter {
         };
 
         if !self.use_separate_crates {
-            formatted = formatted
-                .replace("icu_", "icu::")
-                .replace("icu::provider", "icu_provider")
-                .replace("icu::locale_core", "icu_locale_core")
-                .replace("icu::pattern", "icu_pattern");
+            if formatted[..300].contains("macro_rules!") || formatted[..100].contains("include!") {
```

Comment:
Nit: Please document why this sniffing works. Why 300 bytes, why `macro_rules!` and `include!` as opposed to anything else you could sniff.